src/app/Tests/AccountControllerTests.cs

```diff
@@ -0,0 +1,614 @@
+using Api.Controllers;
+using Api.Core;
+using Api.Core.DAL.Documents.User;
+using Api.Core.DAL.Repositories;
+using Api.Core.Interfaces.Services.Document;
+using Api.Core.Interfaces.Services.Infrastructure;
+using Api.Core.Services.Document.Models;
+using Api.Core.Services.Infrastructure.Models;
+using Api.Core.Settings;
+using Api.Core.Utils;
+using Api.Models.Account;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Moq;
+using Xunit;
+using ForgotPasswordModel = Api.Models.Account.ForgotPasswordModel;
+
+namespace Tests
+{
+    public class AccountControllerTests
+    {
+        private readonly Mock<IEmailService> _emailService;
+        private readonly Mock<IUserService> _userService;
+        private readonly Mock<ITokenService> _tokenService;
+        private readonly Mock<IAuthService> _authService;
+        private readonly Mock<IWebHostEnvironment> _environment;
+        private readonly Mock<IOptions<AppSettings>> _appSettingsOptions;
+        private readonly Mock<IGoogleService> _googleService;
+        private readonly AppSettings _appSettings;
+
+        public AccountControllerTests()
+        {
+            _emailService = new Mock<IEmailService>();
+            _userService = new Mock<IUserService>();
+            _tokenService = new Mock<ITokenService>();
+            _authService = new Mock<IAuthService>();
+            _environment = new Mock<IWebHostEnvironment>();
+            _appSettingsOptions = new Mock<IOptions<AppSettings>>();
+            _googleService = new Mock<IGoogleService>();
+
+            _appSettings = new AppSettings
+            {
+                WebUrl = "http://test.com",
+                LandingUrl = "http://test-landing.com"
+            };
+        }
+
+        [Fact]
+        public async void SignUpShouldReturnBadRequestWhenUserAlreadyExist()
+        {
+            // Arrange
+            var model = new SignUpModel
+            {
+                Email = "sample@sample.com"
+            };
+
+            _userService.Setup(service => service.FindByEmailAsync(model.Email))
+                .ReturnsAsync(new User());
+
+            
+            var controller = CreateInstance();
+
+            // Act
+            var result = await controller.SignUpAsync(model);
+
+            // Assert
+            Assert.IsAssignableFrom<BadRequestResult>(result);
```

Comment:
Maybe, it's better to compare result type with assert type only - I mean, not to check if it's derived? I tested this assert, it also works:
```
Assert.IsType<BadRequestResult>(result);
```